### DISCUSSION TITLE
Document Previews not showing

### DISCUSSION THREAD

**Discussion by avanrossum** [pimcore/pimcore#15834] [Q&A / Support] [answered]
Hello everyone - currently, we're using Community Edition, and I'm wondering if something we're seeing is a limitation of Community, or if i have something configured incorrectly.

No view / preview options are available for Docx, xls, ppt, etc, and thumbnails are not generating - everything is a broken image - but pdfs do preview. Jpg / png etc render thnumbnails just fine, as do text / rtf documents

Libreoffice is installed, and is able to be run by the www-data user

This is running on an apache environment on an Ubuntu server.  Everything is installed / enabled other than Redis

Thank you!
![CleanShot 2023-08-28 at 15 16 50](https://github.com/pimcore/pimcore/assets/25301314/7760cbdf-dc03-4af3-8240-5b848e839cd9)

**Comment by kingjia90** [comment]
Hello,

Is the[ maintenance job](https://pimcore.com/docs/pimcore/10.6/Development_Documentation/Getting_Started/Installation.html#page_5-Maintenance-Cron-Job) active and running? see also [docs](https://pimcore.com/docs/pimcore/10.6/Development_Documentation/Assets/Working_with_Thumbnails/Document_Thumbnails.html#:~:text=Please%20be%20aware%20that%20the%20processing%20of%20thumbnails%20for%20documents%20is%20done%20asynchronously%20as%20part%20of%20the%20maintenance%20job.)

If yes, then other info that would help, is it on Pimcore 10.x or 11.x? Is Ghostscript also installed?

**Comment by avanrossum** [comment]
Oh - sorry - yes the maintenance job is (supposedly) running. I’m getting
image thumbnails for jpg etc and I wasn’t initially.

I’ve set it up with supervisor and with the -vv flag I’ve seen changes
happen.

Currently running pimcore 11

**Reply by kingjia90** [reply, marked as answer]
Can you please try `bin/console messenger:consume pimcore_asset_update`? 
Also please keep an eye on the logs, maybe LibreOffice/Ghostscript is silently failing.

**Reply by avanrossum** [reply]
Thank you! That appears to be doing something.  Looks like i need to add that, pimcore_asset_update, to the other items in the cron / supervisor task.

It's quite busy - and there are a lot of assets - so we will see soon hopefully if this resolved it.

FWIW i did not see any mention of "pimcore_asset_update" int eh 11.x docs, but i might just have missed it somehow.

Thanks again!

**Reply by avanrossum** [reply]
Update: This really seems to be solving it.  I will mark as answer shortly.  Just to be sure - should i add that to the cron / supervisor?

Thank you!

**Reply by kingjia90** [reply]
I've provided a PR to improve the docs a little bit https://github.com/pimcore/pimcore/pull/15836 , i confess that it confused me too at the first times as well, the `pimcore_asset_update` availability is only "hinted"  when typing a wrong value for `messenger:consume` and is ~not mentioned anywhere in the docs~ mentioned as plain text but not in the [example code frame](https://github.com/pimcore/pimcore/blob/f6436cac11dadd6ccb568ea9b4474ce3d144b16e/doc/01_Getting_Started/00_Installation/01_Webserver_Installation.md?plain=1#L94). 
Didn't dig too much, but looks like it was split from other ones at some point and needs now to explicitly defined and/or was intended to be optional.


The official recommendation is to set the own worker accordingly and depending on the project needs, and follow the Symfony guidelines.

Locally, i just do all 4 in one worker, and an heads-up that in this way these queue (eg. pimcore_core pimcore_maintenance pimcore_image_optimize pimcore_search_backend_message pimcore_asset_update) are running in sequence (first processes all the first queue,  then the second, third and so on), but is not recommended for [production](https://symfony.com/doc/current/messenger.html#deploying-to-production)

**Reply by avanrossum** [reply]
Perfect - thank you - marking the answer as solved!
